demos/3d_spherical/3d_spherical.py

```diff
@@ -103,7 +103,7 @@
 
 energy_solver = EnergySolver(T, u, approximation, delta_t, ImplicitMidpoint, bcs=temp_bcs)
 stokes_solver = StokesSolver(z, T, approximation, bcs=stokes_bcs,
-                             cartesian=False,
+                             cartesian=False, J="constant",
```

Comment:
Not a big fan of overloading existing arguments with new meanings depending on type. Providing a custom Jacobian is clearly a separate concern from informing the solver whether the value of the Jacobian is going to be the same in every solve. What happens if we want to provide a custom Jacobian that "is constant"? I would just add a `constant_jacobian` logical argument - using the same name as in Firedrake so people familiar with that in firedrake may recognize it

Reply:
Agreed! I have now switched to having `constant_jacobian` for `StokesSolver`.